controller/controller.go

```diff
@@ -305,6 +327,36 @@ func (c controller) Destroy(req Request) (*task.Task, error) {
 	return taskObject, nil
 }
 
+func (c controller) Update(req Request, opts UpdateOptions) (*task.Task, error) {
+	action := func() error {
+		req, ok, err := c.GroupNeedsUpdate(req)
+		if err != nil {
+			return maskAny(err)
+		}
+
+		if !ok {
+			// Group does not need to be updated. Do nothing.
+			return maskAnyf(updateNotAllowedError, "units already up to date")
+		}
+
+		err = c.UpdateWithStrategy(req, opts)
+		if err != nil {
+			return maskAny(err)
+		}
+
+		// TODO retry operations
```

Comment:
`UpdateWithStrategy` should probably handle all the retries internally. No need for an extra TODO here IMO